nixos/modules/services/torrent/deluge.nix

```diff
@@ -20,9 +60,93 @@ in {
             Number of files to allow deluged to open.
           '';
         };
+
+        coreConf = mkOption {
+          type = types.attrs;
+          example = {
+            download_location = "/srv/torrents/";
+            max_upload_speed = "1000.0";
+            share_ratio_limit = "2.0";
+            allow_remote = true;
+          };
+          description = ''
+            Deluge core configuration for the core.conf file. Only has
+            an effect when <option>services.deluge.declarative</option>
+            is set to <literal>true</literal>.
+            String values must be quoted, integer and boolean values must not.
+          '';
+        };
+
+      declarative = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to use a declarative deluge configuration.
+          Only if set to <literal>true</literal>, the options
+          <option>services.deluge.coreConf</option>,
+          <option>services.deluge.listenPortRange</option> and
+          <option>services.deluge.auth</option> will be
+          applied.
+        '';
       };
 
-      deluge.web.enable = mkEnableOption "Deluge Web daemon";
+        port = mkOption {
+        type = types.ints.u16;
```

Comment:
No need to introduce a separate option for this, it is already in `core.conf` as `daemon_port`. And again, setting this to a default would override the one in `core.conf`, which might be stateful, so we can't do that.

Reply:
It is actually needed; `daemon_port` is the port used for RPC while `services.deluge.web.port` is for the port for the web ui.
Now the default is not to overwrite the web ui port unless `services.deluge.web.port` is defined.